open-codegen/opengen/tcp/solver_status.py

```diff
@@ -0,0 +1,14 @@
+class SolverStatus:
```

Comment:
How about the following implementation?

```python
class SolverStatus:
    
    """Class for storing the status of the solver."""
    def __init__(self, status):
        """Constructs instance of <code>SolverStatus</code>

        Args:
            status: dictionary containing solver status attributes

        Returns:
            New instance of <code>SolverStatus</code>
        """

        for k, v in status.items():
            attribute_name = "__{}".format(k)
            setattr(self, attribute_name, v)

    @property
    def exit_status(self):
        """
        Documentation goes here
        :return: 
        """
        return self.__dict__["__exit_status"]
```

This way methods/properties will be documented. Additionally, we will have control over what functionality becomes available to the user.